libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -302,18 +306,40 @@ public Value parse(XContentParser parser, Value value, Context context) throws I
                         }
                     }
 
+                    // Check if this field is in an exclusive set, if it is then mark
+                    // it as seen.  If the set is already marked, then we have a duplicate
```

Comment:
This doesn't throw an exception though!

Reply:
Oh, I see. It does, but later. Got it. Maybe update the comment. Not sure.

Reply:
I rewrote the comments a bit to make things clearer